common/libs/contract.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+// TODO support events, constructors, fallbacks, array slots, types
+import { sha3, setLengthLeft, toBuffer } from 'ethereumjs-util';
+import { toHex } from 'libs/values';
+import Big from 'big.js';
+
+type ABIType = 'address' | 'uint256' | 'bool';
+
+type ABITypedSlot = {
+  name: string,
+  type: ABIType
+};
+
+type ABIMethod = {
+  name: string,
+  type: 'function',
+  constant: boolean,
+  inputs: ABITypedSlot[],
+  outputs: ABITypedSlot[],
+  // default - false
+  payable?: boolean
+};
+
+export type ABI = ABIMethod[];
+
+function assertString(arg: any) {
+  if (typeof arg !== 'string') {
+    throw new Error('Expected string');
+  }
+}
+
+// Contract helper, returns data for given call
+export default class Contract {
+  abi: ABI;
+  constructor(abi: ABI) {
+    this.abi = abi;
+  }
+
+  getMethodAbi(name: string): ABIMethod {
+    const method = this.abi.find(x => x.name === name);
+    // FIXME
```

Comment:
Any idea what it's calling out? This seems reasonable to me, to explicitly throw if they ask for something wrong.